hoomd/MeshDefinition.h

```diff
@@ -0,0 +1,105 @@
+/*! \file MeshDefinition.h
+    \brief Defines the MeshDefinition class
+ */
+
+#ifdef __HIPCC__
+#error This header cannot be compiled by nvcc
+#endif
+
+#include "BondedGroupData.h"
+#include "IntegratorData.h"
+#include "MeshGroupData.h"
+#include "SystemDefinition.h"
+
+#include <memory>
+#include <pybind11/pybind11.h>
+
+#ifndef __MESH_DEFINITION_H__
+#define __MESH_DEFINITION_H__
+
+namespace hoomd
+    {
+#ifdef ENABLE_MPI
+//! Forward declaration of Communicator
+class Communicator;
+#endif
+
+//! Container class for all data needed to define a mesh system
+/*! MeshDefinition is a big bucket where all of the data for a mesh structure goes.
+    Everything is stored as a shared pointer for quick and easy access from within C++
+    and python without worrying about data management.
+
+    <b>Background and intended usage</b>
+
+    The data structure stored in MeshDefinition is all the bonds and triangles within a mesh.
+    These will need access to information such as the number of particles in the system or
+    potentially some of the per-particle data to allow for dynamic bonding and meshes.
+
+    More generally, any data structure class in MeshDefinition can potentially reference any
+   other, simply by giving the shared pointer to the referenced class to the constructor of the one
+   that needs to refer to it. Note that using this setup, there can be no circular references. This
+   is a \b good \b thing ^TM, as it promotes good separation and isolation of the various classes
+   responsibilities.
+
+    <b>Initializing</b>
+
+    A default constructed MeshDefinition is full of NULL shared pointers. Such is intended to be
+   assigned to by one created by a SystemInitializer.
+
+    Several other default constructors are provided, mainly to provide backward compatibility to
+   unit tests that relied on the simple initialization constructors provided by ParticleData.
+
+    \ingroup data_structs
+*/
```

Comment:
This documentation is not relevant as it was copied and pasted from `SystemDefinition`. Please rewrite and provide a full description of the various mesh data structures and describe how they interact. We especially need to document clearly where the data structures store indices vs tags and which arrays these indicies or tags refer to.